ui-v2/src/components/ui/flow-run-acitivity-bar-graph/index.tsx

```diff
@@ -0,0 +1,292 @@
+import type { components } from "@/api/prefect";
+import { ChartContainer, ChartTooltip } from "@/components/ui/chart";
+import { Link } from "@tanstack/react-router";
+import { cva } from "class-variance-authority";
+import { format, formatDistanceStrict } from "date-fns";
+import { Calendar, ChevronRight, Clock, Rocket } from "lucide-react";
+import { useEffect, useRef, useState } from "react";
+import { Bar, BarChart, Cell, type TooltipProps } from "recharts";
+import {
+	Card,
+	CardContent,
+	CardDescription,
+	CardHeader,
+	CardTitle,
+} from "../card";
+import { StateBadge } from "../state-badge";
+import { TagBadgeGroup } from "../tag-badge-group";
+import { organizeFlowRunsWithGaps } from "./utils";
+
+type CustomShapeProps = {
+	x?: number;
+	y?: number;
+	width?: number;
+	height?: number;
+	radius?: number[];
+	role?: string;
+	flowRun?: EnrichedFlowRun;
+};
+
+const barVariants = cva("gap-1", {
+	variants: {
+		state: {
+			COMPLETED: "fill-green-600",
+			FAILED: "fill-red-600",
+			RUNNING: "fill-blue-700",
+			CANCELLED: "fill-gray-500",
+			CANCELLING: "fill-gray-600",
+			CRASHED: "fill-orange-600",
+			PAUSED: "fill-gray-500",
+			PENDING: "fill-gray-400",
+			SCHEDULED: "fill-yellow-400",
+			NONE: "fill-gray-300",
+		} satisfies Record<components["schemas"]["StateType"] | "NONE", string>,
+	},
+	defaultVariants: {
+		state: "NONE",
+	},
+});
+
+const CustomBar = (props: CustomShapeProps) => {
+	const minHeight = 10; // Minimum height for zero values
+	const {
+		x = 0,
+		y = 0,
+		width = 0,
+		height = minHeight,
+		radius = [0, 0, 0, 0],
+		role,
+		flowRun,
+	} = props;
+
+	return (
+		<g role={role}>
+			<rect
+				data-testid={`bar-rect-${flowRun?.id}`}
+				x={x}
+				y={y + height - Math.max(height, minHeight, width)}
+				width={width}
+				height={Math.max(height, minHeight)}
+				rx={radius[0]}
+				ry={radius[0]}
+				className={barVariants({ state: flowRun?.state_type })}
+			/>
+		</g>
+	);
+};
+
+type EnrichedFlowRun = components["schemas"]["FlowRun"] & {
+	deployment: components["schemas"]["DeploymentResponse"];
+	flow: components["schemas"]["Flow"];
+};
+
+/**
+ * Custom hook to manage tooltip active state with a delayed hide effect. When the tooltip is set to inactive, it will wait for the specified delay before
+ * surrendering control.
+ *
+ * @param initialValue - Initial active state of the tooltip (default: undefined)
+ * @param leaveDelay - Delay in milliseconds before hiding the tooltip after becoming inactive (default: 500ms)
+ * @returns A tuple containing [isActive, setIsActive] where isActive is the current tooltip state
+ *          and setIsActive is a function to update the internal state
+ */
+const useIsTooltipActive = (
+	initialValue: boolean | undefined = undefined,
+	leaveDelay = 500,
+) => {
+	const [internalValue, setInternalValue] = useState<boolean | undefined>(
```

Comment:
q: Is there a differentation between an `undefined` state and `false` state?

Reply:
Yes, `false` will cause the tooltip never to appear, and `undefined` allows `recharts` to control whether or not to display the tooltip. This implementation selectively takes over the state of the tooltip so that users can mouse over the tooltip to click on links to the flow, flow run, or deployment.